### ISSUE TITLE
Bug: Style import are not detected and sorted

### ISSUE THREAD

**Issue by johnny-prescott** [azat-io/eslint-plugin-perfectionist#458]
### Describe the bug

Hi,

I've tried to modify the config quite a few times to make it work but I can't seems to figure out why it doesn't. Everything else seems to be working just fine.

I've used the example from the documentation for my configuration. (As shown below)

Thing is it looks like the import for the scss is completly ignored and I'm not exactly sure why that is. I've tried with both the default "style" rule and making a custom one (like the react one), and both are ignored.

It stays exactly where it was and even split the other group wiht linebreak.

bug or misconfig?

I'm in an NX environement if that has any importance but for my test, I've used eslint CLI with --fix.

thanks.

### Code example

Config (flat):
```
rules: {
			'perfectionist/sort-imports': [
				'error',
				{
					type: 'natural',
					order: 'asc',
					groups: [
						'react',
						'type',
						['builtin', 'external'],
						'internal-type',
						'internal',
						['parent-type', 'sibling-type', 'index-type'],
						['parent', 'sibling', 'index'],
						'object',
						'aliased',
						'style',
						'unknown',
					],
					customGroups: {
						value: {
							react: ['^react$', '^react-.+'],
							aliased: ['^@'],
						},
					},
				},
			],
		},
```

Ran against this set of imports:
```
import { Button, ProgressBar } from 'react-bootstrap';
import { useTranslation } from 'react-i18next';
import React, { MouseEvent, useEffect, useMemo, useState } from 'react';
import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { DynamicIcon } from '@platform/components/dynamic-icon-components';
import { Namespace } from '@platform/router';
import './toast.scss';
import { Toast } from '@platform/types/ui-types';
```

result is:
```
import React, { MouseEvent, useEffect, useMemo, useState } from 'react';
import { Button, ProgressBar } from 'react-bootstrap';
import { useTranslation } from 'react-i18next';

import { IconProp } from '@fortawesome/fontawesome-svg-core';
import { DynamicIcon } from '@platform/components/dynamic-icon-components';
import { Namespace } from '@platform/router';

import './toast.scss';

import { Toast } from '@platform/types/ui-types';
```

### ESLint version

v9.18.0

### ESLint Plugin Perfectionist version

v4.7.0

### Validations

- [x] Read the [docs](https://perfectionist.dev).
- [x] Check that there isn't [already an issue](https://github.com/azat-io/eslint-plugin-perfectionist/issues) that reports the same bug to avoid creating a duplicate.

**Comment by hugop95** [comment]
Hi @johnny-prescott 

You probably need to activate the [`sortSideEffects`](https://perfectionist.dev/rules/sort-imports.html#sortsideeffects) option.

If this doesn't solve your problem, can you please write down the expected result for example?

**Comment by azat-io** [comment]
Closed due to inactivity .
